bpf/lib/nodeport.h

```diff
@@ -755,6 +755,13 @@ static __always_inline int nodeport_lb6(struct __ctx_buff *ctx,
 	if (svc) {
 		const bool skip_l3_xlate = DSR_ENCAP_MODE == DSR_ENCAP_IPIP;
 
+		#if defined(XDP_TRACING) && __ctx_is == __ctx_xdp
+			send_trace_notify(ctx, TRACE_FROM_NETWORK, src_identity, 0,
+					  bpf_ntohs((__u16)svc->backend_id),
```

Comment:
The backend selection happens in the `lb{4,6}_local()`. Perhaps you could move the statement just above `backend_local = __lookup_ip4_endpoint(tuple.daddr);`.

Reply:
Hmm, perhaps I don't understand the issue that well, but I thought the point of #19703 was to capture flow data very early to debug XDP. By placing the tracepoint even further down we lose flows in a lot of conditions (nodeport DSR, NAT_46x64) which we then wouldn't be able to debug.

Perhaps we should reassess our goals and see if this is still the right way to go about it

Reply:
The user is interested for the following trace events:
- Request to a service before the backend selection (i.e., before any packet mangling, but after we know that `svc != NULL`).
- After the request has been mangled (i.e., the DNAT and SNAT xlations have been applied).

For now, in order not to complicate this PR, we can ignore DSR, NAT46x64.

Reply:
What's the conclusion here? The code as it currently stands seems to satisfy the requirements above, no?

Reply:
> What's the conclusion here?

Two traces - after we know that dst is a svc and after svc pkt mangling.

> The code as it currently stands seems to satisfy the requirements above, no?

The second is missing.